modules/account-lib/src/coin/dot/txnSchema.ts

```diff
@@ -0,0 +1,88 @@
+import joi from 'joi';
+import Utils from './utils';
+
+const addressSchema = joi.string().custom((addr) => Utils.isValidAddress(addr));
+
+const proxyTypes = [
+  'Any',
+  'NonTransfer',
+  'Governance',
+  'Staking',
+  'UnusedSudoBalances',
+  'IdentityJudgement',
+  'CancelProxy',
+];
+
+export const BaseTransactionSchema = joi.object({
+  sender: addressSchema.required(),
+  blockNumber: joi.number().required(),
+  blockHash: joi.string().required(),
+  genesisHash: joi.string().required(),
+  metadataRpc: joi.string().required(),
+  specVersion: joi.number().required(),
+  specName: joi.string().valid('kusama', 'polkadot', 'westend', 'statemint', 'statemine').required(),
+  transactionVersion: joi.number().required(),
+  chainName: joi.string().required(),
+  eraPeriod: joi.number().required(),
+  nonce: joi.number().required(),
+  tip: joi.number().optional(),
+});
+
+export const SigningPayloadTransactionSchema = joi.object({
+  eraPeriod: joi.number().optional(),
+  blockHash: joi.string().required(),
+  nonce: joi.number().required(),
+  tip: joi.number().optional(),
+});
+
+export const SignedTransactionSchema = joi.object({
+  sender: addressSchema.required(),
+  nonce: joi.number().required(),
+  eraPeriod: joi.number().optional(),
+  tip: joi.number().optional(),
+});
+
+export const TransferTransactionSchema = joi.object({
+  amount: joi.string().required(),
+  to: addressSchema.required(),
+});
+
+export const StakeTransactionSchema = joi.object({
+  value: joi.string().required(),
+  controller: addressSchema.required(),
+  payee: [
```

Comment:
@gibrangul95 , can you provide a doc or github link, anyhing where did you find info about the payee field.
This is my last structure to check here. 
The others looks good. :+1:

Reply:
The possible options:
https://wiki.polkadot.network/docs/maintain-guides-how-to-nominate-polkadot#step-2-bond-your-dot

The Validation:
1. Building from scratch, should be a string, if we want to specify the account address then it is an object
2. Decoding from tx returns an object with keys and null values. i'e if the payee was `controller` the decoded tx would be returning `payee: { controller: null }` or incase of an account id `payee: { account: "some dot address" }`

Reply:
Excellent Thank you.